debezium-core/src/main/java/io/debezium/transforms/time/TimestampConverter.java

```diff
@@ -0,0 +1,785 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+/*
+Note: All the null values for any fields should be made to optional, if not while validating the value and schema, we would end up getting null value for field error.
+       Going to convert all null value columns into isoptional if they dont contains any data, so to avoid any kinds of (Conversion error: null value for field that is required and has no default value),
+       not just for nestedvalues, but also for struct fields.
+ */
+
+package io.debezium.transforms.time;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.*;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.apache.kafka.connect.transforms.util.SimpleConfig;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.ZoneId;
+import java.util.*;
+import java.util.Date;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * Background:
+ * Debezium currently does not convert the Timestamps/Dates in Mysql DB to formatted String(Timestamp/Date format),
+ * but rather produces timestamp values into long values, as Kafka connect supports.
+ *
+ * This Transformation allows us to change the long values of Timestamp/Date format data into Timestamp/Date formatted String.
+ * and we do this passing some additional connector configuration parameters like below,
+ *
+ * This transform is based on two different set of properties as shown below,
+ * 1. "transforms":"TimestampConv"
+ *    "transforms.TimestampConv.type":"io.debezium.transforms.time.TimestampConverter$Value"
+ *    "transforms.TimestampConv.fields":"<databasename>.<tablename>.<after/before>.<nestedfield>-><destinationtype> || <databasename>.<tablename>.<field>-><destinationtype>"
+ *    "transforms.TimestampConv.timestamp.format":"yyyy-MM-dd hh:mm:ss"
```

Comment:
So it seems the format is linked to the type rather than the selected field? I'd suggest to add it to the field config, too, as you might want to use different formats for multiple fields of the same type, e.g. like so:

    transforms.TimestampConv.fields":"path.to.my.field|java.util.String|yyy-MM-dd"

That way you have all the config applying for one field tied together. And you could do it similarly for the type based approach below.